tests/acceptance/features/bootstrap/TusContext.php

```diff
@@ -137,12 +141,48 @@ public function getResourceLocation(string $user, string $resource, string $spac
 				''
 			)
 		);
-		$this->featureContext->theHTTPStatusCodeShouldBe(201, "Expected response status code should be 201");
+	}
 
-		$locationHeader = $this->featureContext->getResponse()->getHeader('Location');
-		if (\sizeof($locationHeader) > 0) {
-			return $locationHeader[0];
+	/**
+	 * send PATCH request to upload resource through TUS
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 *
+	 * @return void
+	 * @throws Exception|GuzzleException
+	 */
+	public function uploadResourceThroughTUS(string $user, string $resource): void {
+		$resourceLocation = $this->featureContext->getResponse()->getHeader('Location');
+		if (\sizeof($resourceLocation) > 0) {
+			$resourceLocation = $resourceLocation[0];
+		} else {
+			throw new \Exception(__METHOD__ . " Location header could not be found");
 		}
-		throw new \Exception(__METHOD__ . " Location header could not be found");
+		$file = \fopen($this->acceptanceTestsDirLocation() . $resource, 'r');
+
+		$this->featureContext->setResponse(
+			HttpRequestHelper::sendRequest(
+				$resourceLocation,
+				"",
+				'HEAD',
+				$user,
+				$this->featureContext->getPasswordForUser($user),
+				[],
+				""
+			)
+		);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200, "Expected response status code should be 200");
```

Comment:
duplication? the code is checking status code here and thru the `then` step too. maybe only checking from the then step is better.

Reply:
This status check is for `HEAD` request that we do before `PATCH` and the checking of status code through `then` step is for `PATCH`